### DISCUSSION TITLE
How do I limit logins so that only users with verified email can login?

### DISCUSSION THREAD

**Discussion by RenFontes** [pocketbase/pocketbase#2117] [Q&A] [answered]
I mean, I see the field is there for users, but not sure how to limit logins to only users that verified their email. 🤔

**Comment by ganigeorgiev** [comment, marked as answer]
You can restrict collections access to only verified users by using a rule like `@request.auth.verified = true`.

If you want to prevent non-verified users to authenticate at all, then at the moment this is possible only via [auth api hook](https://pocketbase.io/docs/event-hooks/#record-auth-api-hooks):

```go
app.OnRecordAuthRequest().Add(func(e *core.RecordAuthEvent) error {
	if !e.Record.Verified() {
		return apis.NewForbiddenError("You need to verify your account first", nil)
	}

	return nil
})
```

_(Collection option to enforce the `verified` status on login from the Admin UI is in my todo and might be added in the future)_

**Reply by RenFontes** [reply]
Thanks a lot! 😁
